src/components/form/date-input/date-input.component.ts

```diff
@@ -173,9 +174,23 @@ export class DateInputComponent implements ControlValueAccessor, Validator, OnIn
     this.propagateChange(this.rawValue);
   }
 
+  public inputFocus() {
+    this.isTouched = false;
+    this.touch();
+  }
+
   public inputBlur() {
-    this.formControl.markAsTouched();
-    this.propagateChange(this.rawValue);
+    this.isTouched = true;
+    this.touch();
+  }
+
+  public touch() {
+    if (this.isTouched) {
```

Comment:
Instead of having a `isTouched` property and a `touch()` function; I think it would be more explicit to remove the property `isTouched` and split the `touch()` function into 2: `touch()` and `untouch()`.